connectd/connectd.c

```diff
@@ -518,6 +599,18 @@ static struct io_plan *connection_in(struct io_conn *conn,
 	return conn_in(conn, &conn_in_arg);
 }
 
+void handle_peer_alt_addr(struct peer *peer, const u8 *msg)
```

Comment:
This should update connectd (whitelisted_peers and the htable) as well as notifying lightningd.  Otherwise it will require a restart to take effect.